docs/docs/resources/pipeline-components/kafka-app.yaml

```diff
@@ -0,0 +1,50 @@
+# Base component for Kafka-based components.
+# Producer or streaming apps should inherit from this class.
+- type: kafka-app # required
+  name: kafka-app # required
+  namespace: namespace # required
+  # `app` can contain application-specific settings, hence  the user is free to
+  # add the key-value pairs they need. The fields that are specific to kafka
+  # apps are marked with `# kafka-app-specific`
+  app: # required
+    streams: # required, kafka-app-specific
+      brokers: ${broker} # required
+      schemaRegistryUrl: ${schema_registry_url}
+    nameOverride: override-with-this-name # kafka-app-specific
+    key1:
+      key4: value3
+    key2: value1
+    key3: value2
+  # Topic(s) from which the component will read input
+  from:
```

Comment:
I would say that this should only be part of streams-app as producers don't have an input

Reply:
Suppose I was to remove it from here, it would seem like it is introduced in `streams-app`'s implementation. This could be confusing if we end up [deleting the inherited configs](https://github.com/bakdata/kpops/pull/112#discussion_r1112751102).

Reply:
If we choose to [delete the inherited configs](https://github.com/bakdata/kpops/pull/112#discussion_r1112751102) then it should be removed from both here and the streams-app, but otherwise, it feels a bit weird to delete it here only.

Reply:
I think it is confusing to leave it here no matter what the decision in the other thread is. For Kafka app only general stuff should be configured like Schema Registry, Brokers, ... but not input topics as not every Kafka app has input topics

Reply:
I would also like to hear your opinions @raminqaf @disrupted

I still think it's best to keep it in, just because it does exist as a field.
What if I leave a comment that because of <explanation>, with normal usage it's unlikely that the field would ever be used? @philipp94831

Reply:
agreed, as a compromise I would propose to empty out the from section, but leave `# from:` as a comment that this is not supported for producers.

Reply:
done